src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -159,6 +215,10 @@ public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysi
         /// </summary>
         public static IForwardingContext CreateForwardingContext(this ForkConnector<IIncomingPhysicalMessageContext, IForwardingContext> forwardingContext, OutgoingMessage message, string forwardingAddress, IIncomingPhysicalMessageContext sourceContext)
         {
+            Guard.AgainstNull(nameof(message), message);
+            Guard.AgainstNull(nameof(forwardingAddress), forwardingAddress);
```

Comment:
Yes the forwarding context should probably validate that. The address cannot be null or empty since that is not a valid queue address